src/__tests__/prompt/BasePrompt.spec.ts

```diff
@@ -73,4 +74,39 @@ describe('BasePrompt', () => {
       expect(cursor).toBe(richCursor)
     })
   })
+  
+  describe('block', () => {
+    it('should return block when block exists on runner', () => {
+      const config: IPromptConfig<any> & IBasePromptConfig = dataset._prompts[0]
+      const
+        ctx = dataset.contexts[1] as IContextInputRequired,
+        runner = new FlowRunner(ctx),
+        prompt = new MessagePrompt(
+          config as IMessagePromptConfig & IBasePromptConfig,
+          '09894745-38ba-456f-aab4-720b7d09d5b3', // first interaction
+          runner)
+
+      expect(prompt.block).toBe(ctx.flows[1].blocks[0])
+    })
+    
+    it.skip('should return block on flow specified by interaction and not necessarily active flow', () => {
+      // first flow, first (+ only) block: 42e635ea-bc57-4c68-a8d6-20f648968bec
+      // first flow: 957a8923-428a-420f-8053-d23927e0eea0
+      
+    })
+    
+    it('should return null when block absent on runner (and not raise)', () => {
+      const config: IPromptConfig<any> & IBasePromptConfig = dataset._prompts[0]
+      const
+        ctx = dataset.contexts[1] as IContextInputRequired,
+        runner = new FlowRunner(ctx),
+        prompt = new MessagePrompt(
+          config as IMessagePromptConfig & IBasePromptConfig,
+          '09894745-38ba-456f-aab4-720b7d09d5b3', // first interaction
```

Comment:
No change required for this, it's just different from how I tend to approach, and that's fine, but I thought I would let you know in case you like the idea. Usually, if I have a comment like this, it means I should extract something. In this case, I would have done a "refactor->extract to variable" for the value `'09894745-38ba-456f-aab4-720b7d09d5b3'`, and I would have extracted it to a variable with a name: `firstInteractionUuid`, so that the code is self documenting without the need for a comment.

Reply:
:) 

I'm looking forward to begin using builders for our test data. I've been "building" these up as part of `flow-builder` work!